### PR TITLE
[WFLY-14121] mod_cluster undertow context is not started if server is

### PR DESCRIPTION
**Pull request by spyrkob**
Issue https://issues.redhat.com/browse/WFLY-14121

### CODE REVIEW

**Comment by spyrkob** [comment]
@rhusar could you take a look at this when you have a moment? I put this in draft because it needs https://github.com/modcluster/mod_cluster/pull/511 and https://github.com/undertow-io/undertow/pull/1102

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/modcluster/StartWorkersInSuspendedModeTestCase.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import static org.jboss.as.controller.client.helpers.ClientConstants.READ_RESOURCE_OPERATION;
+import static org.jboss.as.controller.client.helpers.ClientConstants.RECURSIVE;
+import static org.jboss.as.controller.client.helpers.ClientConstants.STATUS;
+import static org.jboss.as.controller.client.helpers.ClientConstants.SUCCESS;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.jboss.as.test.integration.management.util.ModelUtil.createOpNode;
+
+/**
+ * @author
+ */
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/modcluster/StartWorkersInSuspendedModeTestCase.java:54]: Incomplete Javadoc.

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/modcluster/StartWorkersInSuspendedModeTestCase.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            Set up balancer using deprecated proxy-list option in modcluster subsystem
+             */
+            this.builder
+               .node(NODE_1)
+               .setup("/subsystem=modcluster/mod-cluster-config=configuration:write-attribute(name=advertise,value=false)")
+               .setup("/socket-binding-group=standard-sockets/remote-destination-outbound-socket-binding=proxy1:add(host=localhost,port=8590)")
+               .setup("/subsystem=modcluster/mod-cluster-config=configuration:list-add(name=proxies,value=proxy1)")
+               .teardown("/subsystem=modcluster/mod-cluster-config=configuration:list-remove(name=proxies,value=proxy1)")
+               .teardown("/socket-binding-group=standard-sockets/remote-destination-outbound-socket-binding=proxy1:remove()")
+               .teardown("/subsystem=modcluster/mod-cluster-config=configuration:write-attribute(name=advertise,value=true)")
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/modcluster/StartWorkersInSuspendedModeTestCase.java:128]: The `/subsystem=modcluster/mod-cluster-config=configuration` is legacy resource – which still works due to  compatibility – but new tests should use current resources (i.e. `/subsystem=modcluster/proxy=default`).

**Review by rhusar — CHANGES_REQUESTED** [review]

**Comment by rhusar** [comment]
I am realizing the test still fails even with mod_cluster upgrade (https://github.com/wildfly/wildfly/pull/14912) since it requires fix for undertow as well (https://github.com/undertow-io/undertow/pull/1263).

```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 43.839 s <<< FAILURE! - in org.jboss.as.test.clustering.cluster.modcluster.StartWorkersInSuspendedModeTestCase
[ERROR] testWorkerNodeIsRegisteredStopped(org.jboss.as.test.clustering.cluster.modcluster.StartWorkersInSuspendedModeTestCase)  Time elapsed: 43.576 s  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[success]> but was:<[failed]>
	at org.junit.Assert.assertEquals(Assert.java:117)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at org.jboss.as.test.clustering.cluster.modcluster.StartWorkersInSuspendedModeTestCase.assertWorkerNodeContextIsStopped(StartWorkersInSuspendedModeTestCase.java:111)
	at org.jboss.as.test.clustering.cluster.modcluster.StartWorkersInSuspendedModeTestCase.testWorkerNodeIsRegisteredStopped(StartWorkersInSuspendedModeTestCase.java:84)
```

Nevertheless, this still probably works with the native implementation which doesn't suffer from UNDERTOW-1883, so in theory we could add the test disabled until a version of Undertow with the fix is pulled in.

**Comment by rhusar** [comment]
Also, please fix the Jira link in the description – those must be WFLY issues.

**Comment by spyrkob** [comment]
@rhusar the Undertow changes are merged now and the tests on this PR passed. I updated the linked JIRA as well.

**Comment by gaol** [comment]
@rhusar may I know if this can be merged if it is OK? thanks.

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/modcluster/StartWorkersInSuspendedModeTestCase.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        Assert.assertEquals(SUCCESS, modelNode.get(OUTCOME).asString());
+        Assert.assertEquals("stopped", modelNode.get(RESULT).get(STATUS).asString());
+    }
+
+    static class ServerSetupTask extends CLIServerSetupTask {
+        public ServerSetupTask() {
+
+            /*
+            Set up balancer using deprecated proxy-list option in modcluster subsystem
+             */
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/modcluster/StartWorkersInSuspendedModeTestCase.java:117]: Just a nitpick, this comment is just false. Possibly a copy paste issue.

**Review by rhusar — APPROVED** [review]
I think this is fine.

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/modcluster/StartWorkersInSuspendedModeTestCase.java (hunk 3 above)
Comment (spyrkob) [reply testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/modcluster/StartWorkersInSuspendedModeTestCase.java]: Thanks, removed the comment now. I missed it when changing the setup method earlier on.

**Comment by gaol** [comment]
@rhusar can we merge this if it is OK ? thanks

**Comment by rhusar** [comment]
@gaol Yes :-)

**Pull request merged**